### ISSUE TITLE
Use Java's SecurityManager to block System.exit() from JPype

### ISSUE THREAD

**Issue by eddieh-xlnx** [Xilinx/RapidWright#283]
Apparently there's a way to add a `SecurityManager` that can be used to convert `System.exit()` calls into throwing an exception. Perhaps RW's Python variant could activate that?

_Originally posted by @eddieh-xlnx in https://github.com/Xilinx/RapidWright/pull/262#discussion_r731381737_

**Comment by eddieh-xlnx** [comment]
The `Security Manager` is being [deprecated in Java 17](https://openjdk.java.net/jeps/411) and will be eventually removed.  There is a possibility of [a new API that could catch `System.exit()` calls](https://bugs.openjdk.java.net/browse/JDK-8199704), but no activity on it for a while.  If we do go down this path, it will dispense warning messages and will get progressively harder to use in later versions of Java.

_Originally posted by @clavin-xlnx in https://github.com/Xilinx/RapidWright/pull/262#discussion_r743312762_

**Comment by eddieh-xlnx** [comment]
This is an example of the warning message users would see each time they `import rapidwright`:
```
WARNING: A terminally deprecated method in java.lang.System has been called
WARNING: System::setSecurityManager has been called by com.xilinx.rapidwright.util.PythonSecurityManager (file:/C:/home/workspace/RapidWright/bin/)
WARNING: Please consider reporting this to the maintainers of com.xilinx.rapidwright.util.PythonSecurityManager
WARNING: System::setSecurityManager will be removed in a future release
```

_Originally posted by @clavin-xlnx in https://github.com/Xilinx/RapidWright/pull/262#discussion_r743322167_

**Comment by eddieh-xlnx** [comment]
Fixed by #284.
